--- a/feeds/luci/applications/luci-app-statistics/root/etc/config/luci_statistics
+++ b/feeds/luci/applications/luci-app-statistics/root/etc/config/luci_statistics
@@ -81,7 +81,7 @@ config statistics 'collectd_cpu'
 	option ValuesPercentage '1'
 
 config statistics 'collectd_cpufreq'
-	option enable '0'
+	option enable '1'
 
 config statistics 'collectd_curl'
 	option enable '0'
@@ -152,7 +152,7 @@ config statistics 'collectd_irq'
 	option Irqs '2 3 4 7'
 
 config statistics 'collectd_iwinfo'
-	option enable '1'
+	option enable '0'
 
 config statistics 'collectd_load'
 	option enable '1'
@@ -209,7 +209,7 @@ config statistics 'collectd_netlink'
 	option AllPortsSummary '0'
 
 config statistics 'collectd_thermal'
-	option enable '0'
+	option enable '1'
 	option IgnoreSelected '0'
 	option Device ''
 
