diff --git a/target/linux/airoha/dts/an7581-gemtek-xg2010g.dts b/target/linux/airoha/dts/an7581-gemtek-xg2010g.dts
index e6085d5cd0..7722eab466 100644
--- a/target/linux/airoha/dts/an7581-gemtek-xg2010g.dts
+++ b/target/linux/airoha/dts/an7581-gemtek-xg2010g.dts
@@ -246,6 +246,59 @@
 	status = "okay";
 };
 
+&cpufreq {
+	reg = <0x0 0x1fa20000 0x0 0x2c0
+	       0x0 0x1efbe000 0x0 0x800>;
+	reg-names = "chip-scu", "mcucfg";
+	airoha,force-direct-pll;
+};
+
+&cpu_opp_table {
+	opp-1250000000 {
+		opp-hz = /bits/ 64 <1250000000>;
+		required-opps = <&smcc_opp15>;
+	};
+	opp-1300000000 {
+		opp-hz = /bits/ 64 <1300000000>;
+		required-opps = <&smcc_opp16>;
+	};
+	opp-1350000000 {
+		opp-hz = /bits/ 64 <1350000000>;
+		required-opps = <&smcc_opp17>;
+	};
+	opp-1400000000 {
+		opp-hz = /bits/ 64 <1400000000>;
+		required-opps = <&smcc_opp18>;
+	};
+	opp-1450000000 {
+		opp-hz = /bits/ 64 <1450000000>;
+		required-opps = <&smcc_opp19>;
+	};
+	opp-1500000000 {
+		opp-hz = /bits/ 64 <1500000000>;
+		required-opps = <&smcc_opp20>;
+	};
+	opp-1550000000 {
+		opp-hz = /bits/ 64 <1550000000>;
+		required-opps = <&smcc_opp21>;
+	};
+	opp-1600000000 {
+		opp-hz = /bits/ 64 <1600000000>;
+		required-opps = <&smcc_opp22>;
+	};
+};
+
+&cpu_smcc_opp_table {
+	smcc_opp15: opp15 { opp-level = <15>; };
+	smcc_opp16: opp16 { opp-level = <16>; };
+	smcc_opp17: opp17 { opp-level = <17>; };
+	smcc_opp18: opp18 { opp-level = <18>; };
+	smcc_opp19: opp19 { opp-level = <19>; };
+	smcc_opp20: opp20 { opp-level = <20>; };
+	smcc_opp21: opp21 { opp-level = <21>; };
+	smcc_opp22: opp22 { opp-level = <22>; };
+};
+
 &eth {
 	status = "okay";
 
