diff --git a/package/emortal/cpufreq/Makefile b/package/emortal/cpufreq/Makefile
index c71f1f36f2..a251b6c5aa 100644
--- a/package/emortal/cpufreq/Makefile
+++ b/package/emortal/cpufreq/Makefile
@@ -5,7 +5,7 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=cpufreq
-PKG_RELEASE:=1
+PKG_RELEASE:=3
 
 include $(INCLUDE_DIR)/package.mk
 
diff --git a/package/emortal/cpufreq/files/cpufreq.uci b/package/emortal/cpufreq/files/cpufreq.uci
index 7d839d8299..5050c021aa 100644
--- a/package/emortal/cpufreq/files/cpufreq.uci
+++ b/package/emortal/cpufreq/files/cpufreq.uci
@@ -19,6 +19,9 @@ SOC="$(strings '/proc/device-tree/compatible' | tail -n1)"
 
 source "/etc/openwrt_release"
 case "$DISTRIB_TARGET" in
+	"airoha/an7581")
+		uci_write_config 0 ondemand 500000 1400000 10 50
+		;;
 	"bcm27xx/bcm2710")
 		uci_write_config 0 ondemand 600000 "$CPU_MAX_FREQ" 10 50
 		;;
diff --git a/package/emortal/cpufreq/files/cpufreq.init b/package/emortal/cpufreq/files/cpufreq.init
--- a/package/emortal/cpufreq/files/cpufreq.init
+++ b/package/emortal/cpufreq/files/cpufreq.init
@@ -43,9 +43,10 @@ start_service() {
 	do
 		[ -z "$(config_get "$NAME" "governor$i")" ] && return
 
-		write_cpufreq_config "governor$i" "$CPUFREQ_PATH/policy$i/scaling_governor"
+		# Set the limits before a governor can request the highest OPP.
 		write_cpufreq_config "minfreq$i" "$CPUFREQ_PATH/policy$i/scaling_min_freq"
 		write_cpufreq_config "maxfreq$i" "$CPUFREQ_PATH/policy$i/scaling_max_freq"
+		write_cpufreq_config "governor$i" "$CPUFREQ_PATH/policy$i/scaling_governor"
 		if [ "$(config_get "$NAME" "governor$i")" = "ondemand" ]; then
 			write_cpufreq_config "sdfactor$i" "$CPUFREQ_PATH/ondemand/sampling_down_factor"
 			write_cpufreq_config "upthreshold$i" "$CPUFREQ_PATH/ondemand/up_threshold"
